src/BayesFilters/src/directional_statistics.cpp

```diff
@@ -22,13 +22,10 @@ MatrixXd bfl::directional_statistics::directional_sub(const Ref<const MatrixXd>&
 }
 
 
-double bfl::directional_statistics::directional_mean(const Ref<const VectorXd>& a, const Ref<const VectorXd>& w)
+VectorXd bfl::directional_statistics::directional_mean(const Ref<const MatrixXd>& a, const Ref<const VectorXd>& w)
 {
-    MatrixXcd complex_a(a.rows(), a.cols());
-
-    complex_a.imag() = a;
-
-    std::complex<double> phasor = (w.array() * complex_a.array().exp()).sum();
-
-    return std::atan2(phasor.imag(), phasor.real());
+    /* For each row i of the matrix a,
+       the method computes the sum of exponentials sum(w_{k} * e^(j*a_{ik})) where j is the imaginary unit
+       and then extract the phase angle using the method arg(x). */
+   return ((std::complex<double>(0.0, 1.0) * a).array().exp().matrix() * w).array().arg();
```

Comment:
With this implementation, we require that a vector of angles should be a row vector. Am I wrong? (See also the comment above). Unless this assumptions is convenient for particular use cases or constraints, I would really prefer to have a column vector or column-stacked column vectors for, respectively, single and multiple directiona mean computations..

Reply:
You are right. The idea was to consider a vector as a state vector containing angles and then performing a weighted mean of the columns. This can be used for example when doing evaluation of sigma points mean in the form of
```
mean = directional_mean(sigma_points, weight)
```
where `sigma_points` is a matrix containing vectors containing angles and `weight` is the usual vector containing weights.
However this may be too state vector-oriented.

Reply:
I see and I'm ok with this kind of assumption. Let's accept this 👍 

However, I would recommend to update the code checking if the user passes a single column vector instead of a Matrix. In that case I would consider doing the mean of the column vector. What do you think about it? Otherwise we just leave the function as it is.

Reply:
Sorry, I resolved the conversation by mistake.

Maybe we can switch to the other representation you suggested because it is more clear for the user, no doubt. The code does not change too much. It suffices to use `a.transpose()` instead of `a`.

Reply:
I suggest to do the following: if the user provides a column vector we return it straight away irrespective of the dimension and the value/dimension of the weights.
Let's keep it simple. What do you think?

Reply:
As long as `a.cols() == w.size()` also the special case of having `a` with one column only is handled and results in `w(0) * a.col(0)`. We may insert a check for the dimensionality and returns a `std::pair<bool, VectorXd>`? But i would prefer to lever the internal checks performed by Eigen.

Reply:
> I suggest to do the following: if the user provides a column vector we return it straight away irrespective of the dimension and the value/dimension of the weights.
Let's keep it simple. What do you think?

When I aswered, your answer was not there. I agree on it!